tests/unit/lib/SGMatrix_unittest.cc

```diff
@@ -519,3 +519,64 @@ TEST(SGMatrixTest, from_eigen3)
 	for (int32_t i=0; i<nrows*ncols; i++)
 		EXPECT_EQ(eigen_mat(i), sg_mat[i]);
 }
+
+TEST(SGMatrixTest, equals)
+{
+	const index_t size=10;
+	SGMatrix<float32_t> mat, copy;
+
+	EXPECT_TRUE(mat.equals(mat));
+	EXPECT_TRUE(mat.equals(copy));
+
+	mat=SGMatrix<float32_t>(size, size);
+	CMath::init_random(100);
+	for (uint64_t i=0; i<mat.size(); ++i)
+		mat.matrix[i]=CMath::randn_float();
+
+	EXPECT_TRUE(mat.equals(mat));
+	EXPECT_FALSE(mat.equals(copy));
+
+	copy=SGMatrix<float32_t>(size, size);
```

Comment:
just allocating the size doesn't mean we're going to read from the memory. segfault otherwise.